runtime/src/iree/hal/drivers/hip/hip_allocator.c

```diff
@@ -432,6 +444,11 @@ static iree_status_t iree_hal_hip_allocator_allocate_buffer(
       iree_hal_buffer_release(buffer);
     }
   }
+
+  status = iree_status_join(
+      status,
+      IREE_HIP_CALL_TO_STATUS(allocator->symbols, hipCtxPopCurrent(NULL)));
+
```

Comment:
this can be a tail call here and below (and in any other file doing this) to make it clearer that it should be the last thing that happens in the function - by not being adjacent to the return it's possible for someone to insert new logic following it and we want that to be an exception.

```suggestion
  return iree_status_join(
      status,
      IREE_HIP_CALL_TO_STATUS(allocator->symbols, hipCtxPopCurrent(NULL)));
```